e2e/test/scenarios/collections/uploads.cy.spec.js

```diff
@@ -281,16 +281,12 @@ function uploadFile(testFile, valid = true) {
       cy.findByText(testFile.humanName);
     });
 
-    cy.findByRole("status").within(() => {
-      cy.findByText("Start exploring").click();
-    });
+    cy.findAllByRole("status").findByText("Start exploring").click();
```

Comment:
This one (and possibly the one above) doesn't work when testing locally.

I've tested with this simple test

```ts
      cy.signInAsAdmin();

      cy.visit("/");
      cy.findByText("Our analytics").should("be.visible");
      cy.findAllByRole("link")
        .should("contain", "Our analytics")
        .and("contain", "Your personal collection");
      // here Cypress detects 5-6 links in my case, but it seems to try to find `Our analytics` only on the first found link
      // it doesn't seem to iterate through all the possibilities to find the matching text.
      cy.findAllByRole("link").findByText("Our analytics").click();
```

I found a (hacky?) solution that could work
```ts
      cy.signInAsAdmin();

      cy.visit("/");
      cy.findByText("Our analytics").should("be.visible");
      cy.findAllByRole("link")
        .should("contain", "Our analytics")
        .and("contain", "Your personal collection");
      cy.findAllByRole("link").each(([$link]) => {
        if ($link.innerText.includes("Our analytics")) {
          cy.wrap($link).as("ourAnalyticsLink");
        }
      });
      cy.get("@ourAnalyticsLink").should("be.visible").click();
```

This works by iterating over all `link` (`status` in your case), and trying to find the element using `innerText` since we can't use any Cypress command to find an element because we don't want the element that doesn't contain our target text to fail as `find*By*` command would fail immediately when the element is not found.

After we found the element we put it in an alias `cy.wrap().as(alias)` then refer to that later on.

You could try to adjust this to your situation. Although this solution isn't elegant, I couldn't think of any other way to achieve this yet as Cypress doesn't [encourage conditional testing.](https://docs.cypress.io/guides/core-concepts/conditional-testing)

Reply:
Thanks for the analysis once again.
This should be slightly less verbose hack that should work in all cases.
https://github.com/metabase/metabase/pull/35579/commits/ccbc68a20a8a04a49448e60ccb73f1b812871b40